src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -261,18 +261,42 @@ private static string GetCharset()
             return null;
         }
 
-        /// <summary>Outputs the format string evaluated and parameterized with the color.</summary>
-        /// <param name="foreground">true for foreground; false for background.</param>
-        /// <param name="color">The color to store into the field and to use as an argument to the format string.</param>
-        private static void ChangeColor(bool foreground, ConsoleColor color)
+        /// <summary>
+        /// Refreshes the foreground and background colors in use by the terminal by resetting
+        /// the colors and then reissuing commands for both foreground and background, if necessary.
+        /// Before doing so, the <paramref name="toChange"/> ref is changed to <paramref name="value"/>
+        /// if <paramref name="value"/> is valid.
+        /// </summary>
+        private static void RefreshColors(ref ConsoleColor toChange, ConsoleColor value)
         {
-            // Get the numerical value of the color
-            int ccValue = (int)color;
-            if ((ccValue & ~0xF) != 0)
+            if (((int)value & ~0xF) != 0 && value != ConsoleColor.Unknown)
             {
                 throw new ArgumentException(SR.Arg_InvalidConsoleColor);
             }
 
+            lock (Console.Out)
```

Comment:
We are.  But we previously changed this to lock on something private, and it ended up breaking scenarios, so we changed back as the lesser of two evils.  Customers lock on Console.Out in order to synchronize coloring with writing, so you can do things like:
```C#
lock (Console.Out)
{
    Console.ForegroundColor = ConsoleColor.Yellow;
    Console.WriteLine("yellow");
    Console.ForegroundColor = ConsoleColor.White;
}
```
and only "yellow" will be written in yellow, and not other text being written by other threads.  We have no other mechanism in the API for such synchronization, so we reverted. https://github.com/dotnet/corefx/pull/2855